circuit/src/integer/rns.rs

```diff
@@ -234,146 +356,187 @@ impl RnsParams<Fq, Fr, 4, 68> for Bn256_4_68 {
 		[limb0, limb1, limb2, limb3]
 	}
 
-	fn residues(n: &[Fr; 4], t: &[Fr; 4]) -> Vec<Fr> {
-		let lsh1 = Self::left_shifters()[1];
-		let rsh2 = Self::right_shifters()[2];
+	// TODO: Move outside Rns -- Use just BigUint as output
+	/// Computes the inverse of the [`BigUint`] as an element of the Wrong
+	/// field. Returns `None` if the value cannot be inverted.
+	fn invert(input: BigUint) -> Option<Integer<Fq, Fr, 4, 68, Bn256_4_68>> {
+		let a_w = big_to_fe::<Fq>(input);
+		let inv_w = a_w.invert();
+		inv_w.map(|inv| Integer::<Fq, Fr, 4, 68, Bn256_4_68>::new(fe_to_big(inv))).into()
+	}
+}
```

Comment:
Looks like we could move this function into the trait as well.

And this TODO can be removed - I think it's not such a good idea anymore